admiral/pkg/clusters/types.go

```diff
@@ -79,13 +80,21 @@ type GlobalTrafficHandler struct {
 	RemoteRegistry *RemoteRegistry
 }
 
+type RolloutHandler struct {
+	RemoteRegistry *RemoteRegistry
+}
+
 type globalTrafficCache struct {
 	//map of global traffic policies key=environment.identity, value: GlobalTrafficPolicy object
 	identityCache map[string]*v1.GlobalTrafficPolicy
 
 	//map of dependencies. key=namespace.globaltrafficpolicy name. value Deployment object
 	dependencyCache map[string]*k8sAppsV1.Deployment
 
+
+	//map of dependencies. key=namespace.globaltrafficpolicy name. value Rollout object
+	dependencyRolloutCache map[string]*argo.Rollout
```

Comment:
Seems to me like all of the logic for the two caches is the same. Could they be combined into one? (Using an interface that's implemented by both Deployments and Rollouts)?